src/main/webapp/src/components/autocompleteTextField.js

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { withStyles } from "@material-ui/styles";
+import TextField from '@material-ui/core/TextField';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import Typography from '@material-ui/core/Typography';
+import throttle from 'lodash/throttle';
+import axios from "axios";
+
+/**
+ * Change TextField outline border default colors.
+ */
+const StyledTextField = withStyles({
+  root: {
+    "& label.Mui-focused": {
+      color: "black",
+    },
+    "& .MuiInput-underline:after": {
+      borderBottomColor: "#733F94",
+    },
+    "& .MuiOutlinedInput-root": {
+      "&.Mui-focused fieldset": {
+        borderColor: "#733F94",
+      },
+    },
+  },
+})(TextField);
+
+export default function AutocompleteTextField(props) {
+  const [value, setValue] = React.useState(props.value);
+  const [options, setOptions] = React.useState([]);
+
+  const fetch = React.useMemo(
+    () =>
+      throttle((request, callback) => {
+        axios
+          .post(`/autocomplete?type=${props.type}&searchString=${request}`)
+          .then((result) => result.data)
+          .then((options) => callback(options));
+      }, 200),
+    [],
+  );
+
+  React.useEffect(() => {
+    let active = true;
+
+    if (value === '') {
+      return undefined;
+    }
```

Comment:
You can move this before `let active = true;`. Also, probably `return;` would be enough. Please add a comment explaining why we're returning early when value is empty.